packages/relay/tests/lib/eth.spec.ts

```diff
@@ -3033,6 +3033,55 @@ describe('Eth calls using MirrorNode', async function () {
     expect(gas).to.equal(EthImpl.numberTo0x(defaultGasOverride));
   });
 
+  it('eth_estimateGas with contract revert returns the JsonRpcError', async function () {
+    const transaction = {
+      from: "0x05fba803be258049a27b820088bab1cad2058871",
+      data: "0x60806040523480156200001157600080fd5b50604051620019f4380380620019f48339818101604052810190620000379190620001fa565b818181600390816200004a9190620004ca565b5080600490816200005c9190620004ca565b5050505050620005b1565b6000604051905090565b600080fd5b600080fd5b600080fd5b600080fd5b6000601f19601f8301169050919050565b7f4e487b7100000000000000000000000000000000000000000000000000000000600052604160045260246000fd5b620000d08262000085565b810181811067ffffffffffffffff82111715620000f257620000f162000096565b5b80604052505"
+    };
+    const id = uuid();
+    web3Mock.onPost('contracts/call', {...transaction, estimate: true}).reply(400, {
+      "_status": {
+        "messages": [
+          {
+            "data": "solidity require failed!",
+            "detail": "Generic detailed error message",
+            "message": "Generic error message"
+          }
+        ]
+      }
+    });
+
+    const contractRevert = predefined.CONTRACT_REVERT("solidity require failed!");
+    const result: any = await ethImpl.estimateGas(transaction, id);
+
+    expect(result.code).to.equal(contractRevert.code);
+    expect(result.message).to.equal(contractRevert.message);
```

Comment:
This will never be true as you've mocked the mirror node response with a message that differs from the expected `contractRevert` error message.
If you expect them to be equal then the message should be the same

Reply:
Ah i see it now. The error message is in the data

Reply:
The MirrorNodeClientError constructor parses the incoming message into a message and data and the errorMessage is set using the data.  The errorMessage is returned and later used to set the predefined.CONTRACT_REVERT.

Reply:
That flow is actually problematic because the mirror node error message is actually in the message part of the response. I've opened #1429 which should be addressed first before your PR. I have a feeling that is contributing to the empty errors